fem/tmop.hpp

```diff
@@ -748,6 +802,16 @@ class TMOP_Integrator : public NonlinearFormIntegrator
    // Normalization factor for the limiting term.
    double lim_normal;
 
+   mutable DiscreteAdaptTC *discr_tc;
+
+   // Parameters for finite difference (FD) Gradient & Hessian calculations.
+   int    fdflag;
+   double fdeps;
+   double elemenergy;
```

Comment:
This changes from element to element, so it's better to have it as local variable. It should be added as argument to `GetFDDerivative`, to simplify the logic.